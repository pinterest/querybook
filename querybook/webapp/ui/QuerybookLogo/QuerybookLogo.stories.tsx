import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { QuerybookLogo } from './QuerybookLogo';

export default {
    title: 'Stateless/QuerybookLogo',
    decorators: [centered],
};
export const _QuerybookLogo = (args) => <QuerybookLogo {...args} />;

_QuerybookLogo.args = {
    size: 2,
    withBrandMark: true,
};

_QuerybookLogo.argTypes = {
    size: {
        control: {
            type: 'range',
            min: 1,
            max: 10,
        },
    },
};
