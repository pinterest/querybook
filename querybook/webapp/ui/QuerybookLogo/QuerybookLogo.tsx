import React from 'react';
import styled from 'styled-components';
import { getAppName } from 'lib/utils/global';

const StyledQuerybookLogo = styled.span`
    display: inline-flex;
    align-items: center;
    justify-content: center;

    font-size: ${({ size }) => `${size}rem`};
    .querybook-brandmark {
        width: ${({ size }) => `${size * 1.3}rem`};
        height: ${({ size }) => `${size * 1.3}rem`};
    }

    .querybook-wordmark {
        font-weight: 700;

        letter-spacing: -0.05em;
        padding-right: 0.05em;

        user-select: none;
        color: var(--color-accent);

        background: -webkit-linear-gradient(
            45deg,
            var(--color-accent-text),
            var(--color-accent),
            var(--color-accent-bg)
        );
        background-clip: text;
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
    }
`;

export const QuerybookLogo: React.FC<{
    size?: number;
    withBrandMark?: boolean;
}> = ({ size = 2, withBrandMark }) => (
    <StyledQuerybookLogo size={size}>
        {withBrandMark && (
            <img
                className="querybook-brandmark"
                src={'/static/favicon/favicon.ico'}
            />
        )}
        <span className="querybook-wordmark">{getAppName()}</span>
    </StyledQuerybookLogo>
);
