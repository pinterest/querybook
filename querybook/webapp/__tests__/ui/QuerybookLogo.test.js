import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { QuerybookLogo } from '../../ui/QuerybookLogo/QuerybookLogo';

it('renders without crashing', () => {
    shallow(<QuerybookLogo />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<QuerybookLogo />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot - size', () => {
        const wrapper = shallow(<QuerybookLogo size={8} />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<QuerybookLogo />);
        expect(output).toMatchSnapshot();
    });
});
