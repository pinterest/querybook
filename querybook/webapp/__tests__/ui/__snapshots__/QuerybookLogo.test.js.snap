// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot - size 1`] = `
<QuerybookLogo__StyledQuerybookLogo
  size={8}
>
  <span
    className="querybook-wordmark"
  >
    Querybook
  </span>
</QuerybookLogo__StyledQuerybookLogo>
`;

exports[`matches enzyme snapshots matches snapshot 1`] = `
<QuerybookLogo__StyledQuerybookLogo
  size={2}
>
  <span
    className="querybook-wordmark"
  >
    Querybook
  </span>
</QuerybookLogo__StyledQuerybookLogo>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<span
  className="QuerybookLogo__StyledQuerybookLogo-sc-1aw8dxl-0 cTjbvo"
  size={2}
>
  <span
    className="querybook-wordmark"
  >
    Querybook
  </span>
</span>
`;
